### PR TITLE
Feature/311 filter by participation

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/app/filter/entity.ml [hunk 1]
```diff
@@ -195,6 +198,7 @@ module Key = struct
     | NumInvitations -> "pool_contacts.num_invitations"
     | NumParticipations -> "pool_contacts.num_participations"
     | NumShowUps -> "pool_contacts.num_show_ups"
+    | Participation -> failwith "TODO"
```
Comment (mabiede) [inline pool/app/filter/entity.ml:201]: open TODO

File: pool/app/filter/repo/repo_utils.ml [hunk 2]
```diff
@@ -32,28 +32,85 @@ let filtered_base_condition =
...
+  | Nr n -> add Caqti_type.float n
+  | Option id ->
+    add
+      Caqti_type.string
+      (id |> Custom_field.SelectOption.Id.value |> wrap_in_percentage operator)
+  | Str s -> add Caqti_type.string (wrap_in_percentage operator s)
+;;
+
+(* The subquery does not return any contacts that have shown up at a session of
+   the current experiment. It does not make a difference, it they
```
Comment (mabiede) [inline pool/app/filter/repo/repo_utils.ml:57]:
```suggestion
   the current experiment. It does not make a difference, if they
```

File: pool/app/filter/repo/repo_utils.ml [hunk 3]
```diff
@@ -32,28 +32,85 @@ let filtered_base_condition =
...
+   participated. *)
+let participation_subquery _ dyn operator ids =
```
Comment (mabiede) [inline pool/app/filter/repo/repo_utils.ml:59]: Why is the first input argument skipped?

File: pool/web/view/component/component_filter.ml [hunk 4]
```diff
@@ -50,9 +51,7 @@ let htmx_attribs
   ()
   =
   let target =
-    target
-    |> CCOption.map (fun target ->
-         a_user_data "hx-target" (Format.asprintf "#%s" target))
+    target |> CCOption.map (fun target -> a_user_data "hx-target" target)
```
Comment (mabiede) [inline pool/web/view/component/component_filter.ml:54]:
```suggestion
    target |> CCOption.map (a_user_data "hx-target")
```

File: pool/web/view/component/component_filter.ml [hunk 5]
```diff
@@ -235,7 +306,7 @@ let value_input language input_type ?value () =
              | Lst lst ->
                CCList.filter_map
                  (fun value ->
-                   match value with
+                   match[@warning "-4"] value with
```
Comment (mabiede) [inline pool/web/view/component/component_filter.ml:309]: How many options to add to get rid of the warning suppression?

File: pool/web/view/component/component_filter.ml [hunk 6]
```diff
@@ -255,17 +326,46 @@ let value_input language input_type ?value () =
          language
          multi_select
          field_name
-         ())
+         ()
+     | Key.QueryExperiments ->
+       let current =
+         value
+         |> CCOption.map_or ~default:[] (fun value ->
```
Comment (mabiede) [inline pool/web/view/component/component_filter.ml:333]:
```suggestion
         |> CCOption.map_or ~default:[] (function
```

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/web/view/component/component_filter.ml (hunk 5 above)
Comment (timohuber) [reply pool/web/view/component/component_filter.ml:309]: i think 8 times 5

**Review by mabiede — APPROVED** [review]
merge conflicts, update changelog

**Pull request merged**